### PR TITLE
Switch from glslang to DXC

### PR DESCRIPTION
**Pull request by DethRaid**
This PR switched Nova from glslang to DXC. This gives us much better HLSL support, which is good because HLSL is Nova's main shading language now

This PR has two problems that we need to solve before merging:
* Every file is in LFS now. Only binaries should be in LFS
* I'm using some macros from Windows headers. I'll need to either write our own versions or find some equivalent or something idk

Creating this PR should at least let us review the changes early

### CODE REVIEW

**Comment by DethRaid** [comment]
Removed the Do Not Merge label - Janrupf solved a lot of the issues, we should be ready to merge once we figure out how to make CI happy

**Comment by Janrupf** [comment]
> Removed the Do Not Merge label - Janrupf solved a lot of the issues, we should be ready to merge once we figure out how to make CI happy

Not sure about making CI happy, in theory the changes to the workflow included in this PR should make CI happy, but since the changes are not merged yet and GH always takes the workflow from master, we can't really test it. (It currently fails due to LFS)

**Comment by DethRaid** [comment]
Ugh

File: docs/Shader compilation woes.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+* Checks for errors, printing any if they exist
+* Applies defaults for the shader entry point name
+* Initializes a `dxc::DxcSupport` DLL object
+* Creates a `dxc::DxcContext` with the options and support DLL object
+* Preprocesses the file
+* Compiles the fine
+* Checks for and prints compilation errors
+
+`dxc::DxcContext::Compile`
+* Of lawdy
```
Comment (cwfitzgerald) [inline docs/Shader compilation woes.md]: _Oh_ lawdy 😉

File: docs/Shader compilation woes.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# What does DXC do?
```
Comment (cwfitzgerald) [inline docs/Shader compilation woes.md]: Ahhh filename has spaces. Cursed file.

File: docs/Shader compilation woes.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+`int main`
+* Calls `dxc::main`
+
+`dxc::main`
+* Inits their malloc
+* Inits th arguments table
```
Comment (cwfitzgerald) [inline docs/Shader compilation woes.md]: the

File: docs/Shader compilation woes.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+Shader compilation woes
```
Comment (cwfitzgerald) [inline docs/Shader compilation woes.txt]: What is this file?

File: external/dxc/GIT-COMMIT.txt [hunk 5]
```diff
@@ -0,0 +1 @@
+7a8c4a20da905a736637595fc6111c3ce033ba21
```
Comment (cwfitzgerald) [inline external/dxc/GIT-COMMIT.txt]: Why is the entirety of dxc in this PR?

File: tools/git-hooks/hooks-revision [hunk 6]
```diff
@@ -1 +1 @@
-3630d3d01ccbc7ca73daec58f2a14d1158ae49c0
+7802a58432024132a20d5562cb30a04a040a9fdc
```
Comment (cwfitzgerald) [inline tools/git-hooks/hooks-revision:1]: This file shouldn't be 755 @Janrupf

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
Uh... maybe don't check in all of the dxc wrapper to the repo...

Also some nits

File: external/dxc/GIT-COMMIT.txt (hunk 5 above)
Comment (Janrupf) [reply external/dxc/GIT-COMMIT.txt]: Probably because we are using the binary version, not sure if we need all of those files tho.

Comment (cwfitzgerald) [reply external/dxc/GIT-COMMIT.txt]: Submodule? This is an external repo.

Comment (Janrupf) [reply external/dxc/GIT-COMMIT.txt]: I actually have no idea where @DethRaid has this files from

Comment (cwfitzgerald) [reply external/dxc/GIT-COMMIT.txt]: https://github.com/project-arcana/dxc-wrapper

File: docs/Shader compilation woes.md (hunk 1 above)
Comment (DethRaid) [reply docs/Shader compilation woes.md]: Yeah I fixed

File: external/dxc/GIT-COMMIT.txt (hunk 5 above)
Comment (DethRaid) [reply external/dxc/GIT-COMMIT.txt]: Fixed

**Review by cwfitzgerald — APPROVED** [review]

**Pull request merged**